test/GithubApi.Repositories.test.js

```diff
@@ -0,0 +1,67 @@
+const { StatusCodes } = require('http-status-codes');
+const { expect } = require('chai');
+const axios = require('axios');
+const chai = require('chai');
+const chaiSubset = require('chai-subset');
+const md5 = require('md5');
+
+chai.use(chaiSubset);
+
+const urlBase = 'https://api.github.com';
+const githubUserName = 'aperdomob';
+const company = 'Perficient Latam';
+const name = 'Alejandro Perdomo';
+const repoName = 'jasmine-json-report';
+const desc = 'A Simple Jasmine JSON Report';
+
+describe('Github Api Test', () => {
+  let userResponse;
+
+  before(async () => {
+    userResponse = await axios.get(`${urlBase}/users/${githubUserName}`);
+  });
+
+  it('GET method consumption', () => {
+    expect(userResponse.status).to.equal(StatusCodes.OK);
+    expect(userResponse.data.login).to.equal(githubUserName);
+    expect(userResponse.data.company).to.equal(company);
+    expect(userResponse.data.name).to.equal(name);
+  });
+  describe('Hypermedia', () => {
+    let repository;
+    before(async () => {
+      const repositoryRespose = await axios.get(`${userResponse.data.repos_url}`);
+      repository = repositoryRespose.data.find((Repo) => Repo.name === repoName);
+    });
+    it('GET method for repos', () => {
+      expect(repository.name).to.equal(repoName);
+      expect(repository.private).to.equal(false);
+      expect(repository.description).to.equal(desc);
+    });
+    it('Downloading by zip', async () => {
+      // code to verify the zip download
+      const fileDownload = await axios.get(`${repository.url}/zipball/master`);
+      expect(fileDownload.status).to.equal(StatusCodes.OK);
+      expect(fileDownload.headers).to.contain({ 'content-type': 'application/zip' });
+    });
+    describe('GET repos content', () => {
+      // code to verify the readme and to download and verify the md5
+      let fileDownload;
+      before(async () => {
+        fileDownload = await axios.get(`${repository.url}/contents`);
+      });
+      it('GET README.md file info', () => {
+        expect(fileDownload.data).to.containSubset([{
+          name: 'README.md',
+          path: 'README.md',
+          sha: '360eee6c223cee31e2a59632a2bb9e710a52cdc0'
+        }]);
+      });
+      it('Download README.md file and check md5', async () => {
+        const fileContent = await axios.get(`https://raw.githubusercontent.com/${githubUserName}/${repoName}/master/README.md`);
```

Comment:
if you use a find in the `fileDownload.data` of the response of the previous test, you can get the data of the readme file, and you can use a property called `download_url` to get this URL
![image](https://user-images.githubusercontent.com/10209180/176712427-fb4923f8-1b80-4c81-9044-9ecb93347581.png)